pkg/controllers/state/statenode.go

```diff
@@ -181,26 +176,33 @@ func (in *StateNode) Labels() map[string]string {
 }
 
 func (in *StateNode) Taints() []v1.Taint {
-	// Only consider startup taints until the node is initialized. Without this, if the startup taint is generic and
-	// re-appears on the node for a different reason (e.g. the node is cordoned) we will assume that pods can
-	// schedule against the node in the future incorrectly.
-	ephemeralTaints := scheduling.KnownEphemeralTaints
-	if !in.Initialized() && in.Managed() && in.NodeClaim != nil {
-		ephemeralTaints = append(ephemeralTaints, in.NodeClaim.Spec.StartupTaints...)
-	}
-
 	var taints []v1.Taint
-	if (!in.Registered() && in.NodeClaim != nil) || in.Node == nil {
+	if (!in.Registered() && in.Managed()) || in.Node == nil {
```

Comment:
Worth a comment detailing when a nodeclaim could be not registered but there be a node object and vice versa?